misc/eni-trunking-validator/eni-trunking-validator.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/sparrc/go-ping"
+)
+
+const (
+	imdsEndpoint            = "http://169.254.169.254/latest/meta-data/"
+	v2MetadataEndpoint      = "http://169.254.170.2/v2/metadata"
+	v2StatsEndpoint         = "http://169.254.170.2/v2/stats"
+	containerMetadataEnvVar = "ECS_CONTAINER_METADATA_URI"
+	maxRetries              = 4
+	durationBetweenRetries  = time.Second
+)
+
+func verifyEndPoint(client *http.Client, endPoint string) error {
+	_, err := metadataResponse(client, endPoint)
```

Comment:
Is there no metatdata response that we care to validate?  I see that for all tests we check if there's an error and don't care otherwise.  Is there a case where I would want to validate my expected response?

Reply:
We intentionally steered away from validating the fields in the response as we are already doing it in V3 and metadatavalidator functional tests. The intention with this test is to ensure that end points are reachable as the network request is going through an ENI trunk.